cmd/etcdbackup/abs_writer.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"encoding/base64"
+	"io"
+
+	azstorage "github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/pborman/uuid"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+)
+
+const (
+	// AzureBlobBlockChunkLimitInBytes 100MiB is the limit
+	AzureBlobBlockChunkLimitInBytes = 100 * 1024 * 1024
+)
+
+type blobWriter struct {
+	blob   *azstorage.Blob
+	blocks []azstorage.Block
+}
+
+var _ io.WriteCloser = &blobWriter{}
```

Comment:
for performance you also certainly need to implement io.ReaderFrom and read in at least 1MiB (I'd have thought) chunks.

Reply:
I ended up using an iobuf, i hope that's ok.